net/instaweb/rewriter/rewrite_context.cc

```diff
@@ -1996,13 +1998,28 @@ void RewriteContext::Harvest() {
 void RewriteContext::Render() {
 }
 
+bool RewriteContext::AreOutputsAllowedByCsp(CspDirective role) const {
+  if (Driver()->content_security_policy().empty()) {
+   return true;
+  }
+
+  for (const OutputResourcePtr& o : outputs_) {
+    if (o.get() != nullptr && o->has_hash() && o->has_url() &&
```

Comment:
Hmm, it is a precondition for calling ->url(), though I don't remember off the top of my head when you would not have it.
Maybe I should run the tests with a DCHECK

(That is_url is different only for InlneOutputResource is kinda weird, too)